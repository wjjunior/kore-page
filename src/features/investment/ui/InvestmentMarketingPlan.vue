<template>
  <div>
    <h2
      class="font-hanken-grotesk font-bold text-[32px] leading-[100%] tracking-[1%] text-primary-200 mb-6"
    >
      Performance Marketing Plan
    </h2>

    <div v-if="loading" class="space-y-8">
      <div v-for="i in 3" :key="i" class="animate-pulse">
        <div class="h-8 bg-gray-200 rounded mb-4 w-48"></div>
        <div class="space-y-3">
          <div class="h-4 bg-gray-200 rounded w-full"></div>
          <div
            class="h-4 bg-gray-200 rounded"
            :class="i === 1 ? 'w-3/4' : i === 2 ? 'w-full' : 'w-4/5'"
          ></div>
          <div
            v-if="i > 1"
            class="h-4 bg-gray-200 rounded"
            :class="i === 2 ? 'w-5/6' : 'w-3/4'"
          ></div>
        </div>
      </div>
    </div>

    <div v-else class="flex flex-col gap-6 max-w-4xl">
      <div v-html="props.marketingPlan"></div>
    </div>
  </div>
</template>

<script setup lang="ts">
defineOptions({
  name: "InvestmentMarketingPlan",
});

interface InvestmentMarketingPlanProps {
  loading?: boolean;
  marketingPlan?: string;
}

const props = withDefaults(defineProps<InvestmentMarketingPlanProps>(), {
  loading: false,
  marketingPlan: "",
});
</script>
