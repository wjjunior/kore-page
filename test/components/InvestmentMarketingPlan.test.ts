import { describe, it, expect, vi, beforeEach } from "vitest";
import { mount } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import InvestmentMarketingPlan from "@/features/investment/ui/InvestmentMarketingPlan.vue";

const mountInvestmentMarketingPlan = (props = {}) => {
  return mount(InvestmentMarketingPlan, { props });
};

describe("InvestmentMarketingPlan", () => {
  beforeEach(() => {
    setActivePinia(createPinia());
    vi.clearAllMocks();
  });

  it("renders the section title", () => {
    const wrapper = mountInvestmentMarketingPlan();
    expect(wrapper.find("h2").text()).toBe("Performance Marketing Plan");
  });

  it("shows loading skeletons when loading is true", () => {
    const wrapper = mountInvestmentMarketingPlan({ loading: true });
    const skeletons = wrapper.findAll(".animate-pulse");
    expect(skeletons).toHaveLength(3);
    skeletons.forEach((skeleton) => {
      expect(skeleton.find(".bg-gray-200").exists()).toBe(true);
    });
  });

  it("does not show skeletons when loading is false", () => {
    const wrapper = mountInvestmentMarketingPlan({
      loading: false,
      marketingPlan: "<p>Content</p>",
    });
    expect(wrapper.findAll(".animate-pulse")).toHaveLength(0);
  });

  it("renders marketing plan HTML when loading is false", () => {
    const content = "<p>This is <strong>safe</strong> content</p>";
    const wrapper = mountInvestmentMarketingPlan({
      loading: false,
      marketingPlan: content,
    });

    expect(wrapper.html()).toContain("This is <strong>safe</strong> content");
  });

  it("renders empty state when marketingPlan is not provided", () => {
    const wrapper = mountInvestmentMarketingPlan({ loading: false });
    expect(wrapper.find(".flex.flex-col.gap-6").text()).toBe("");
  });

  it("sanitizes potentially unsafe HTML", () => {
    const unsafeContent = `<p>Safe</p><script>alert('x')</script>`;
    const wrapper = mountInvestmentMarketingPlan({
      loading: false,
      marketingPlan: unsafeContent,
    });

    expect(wrapper.html()).toContain("Safe");
    expect(wrapper.html()).not.toContain("<script>");
  });

  it("renders complex HTML structure correctly", () => {
    const complex = `
      <h3>Marketing Strategy</h3>
      <ul><li>Digital</li><li>Social</li></ul>
      <p><strong>150%</strong></p>
    `;
    const wrapper = mountInvestmentMarketingPlan({
      loading: false,
      marketingPlan: complex,
    });

    expect(wrapper.html()).toContain("Marketing Strategy");
    expect(wrapper.html()).toContain("Digital");
    expect(wrapper.html()).toContain("150%");
  });

  it("has correct default props", () => {
    const wrapper = mountInvestmentMarketingPlan();
    expect(wrapper.props("loading")).toBe(false);
    expect(wrapper.props("marketingPlan")).toBe("");
  });

  it("matches snapshot", () => {
    const wrapper = mountInvestmentMarketingPlan({
      loading: false,
      marketingPlan: "<p>Snapshot content</p>",
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
