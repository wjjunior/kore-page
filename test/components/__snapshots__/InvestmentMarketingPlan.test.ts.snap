// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InvestmentMarketingPlan > matches snapshot 1`] = `
"<div>
  <h2 class="font-hanken-grotesk font-bold text-2xl md:text-3xl lg:text-[30px] xl:text-[32px] leading-[100%] tracking-[1%] text-primary-200 mb-6"> Performance Marketing Plan </h2>
  <div class="flex flex-col gap-6 max-w-full lg:max-w-3xl xl:max-w-4xl">
    <div class="prose prose-sm md:prose prose-gray max-w-none break-words">
      <p>Snapshot content</p>
    </div>
  </div>
</div>"
`;
